rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -0,0 +1,780 @@
+                    raise AssertionError(f'Unexpected AdEx event type: {type(event)}.')
+
+        return ADXStakingEvents(
+            bonds=bonds,
+            unbonds=unbonds,
+            unbond_requests=unbond_requests,
+        )
+
+    @staticmethod
+    def _get_bond_id(
+            identity_address: ChecksumAddress,
+            amount: int,
+            pool_id: HexStr,
+            nonce: int,
+    ) -> HexStr:
+        """Given a LogBond event data, return its `bondId`.
+        """
+        arg_types = ['address', 'address', 'uint', 'bytes32', 'uint']
+        args = [STAKING_ADDR, identity_address, amount, pool_id, nonce]
+        return HexStr(Web3.keccak(Web3().codec.encode_abi(arg_types, args)).hex())
+
+    def _get_identity_address_map(
+            self,
+            addresses: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumAddress, ChecksumAddress]:
+        """Returns a map between the user identity address in the protocol and
+        the EOA/contract address.
+        """
+        return {self._get_user_identity(address): address for address in addresses}
+
+    def _get_tom_pool_incentive(self) -> Optional[TomPoolIncentive]:
+        """Get Tom pool incentive data (staking rewards).
+
+        NB: the APR is the AdEx codebase and website stats APY.
+        https://github.com/AdExNetwork/adex-staking/blob/master/src/actions/actions.js#L86
+
+        TODO: once AdEx rolls out issue #94, check how they handle incentives
+        with multiple channels, for instance how they process `periodEnd`.
+        https://github.com/AdExNetwork/adex-staking/issues/94
+        """
+        fee_rewards = self._get_tom_pool_fee_rewards_from_api()
+        total_staked_amount = FVal('0')
+        total_reward_per_second = FVal('0')
+        period_ends_at = Timestamp(0)
+        apr = FVal('0')
+        for entry in fee_rewards:
+            try:
+                if entry['channelArgs']['tokenAddr'] == A_ADX.ethereum_address:
+                    total_staked_amount += entry['stats']['currentTotalActiveStake']
+                    total_reward_per_second += entry['stats']['currentRewardPerSecond']
+                    # NB: `period_ends_at` is always overwritten
+                    period_ends_at = Timestamp(int(
+                        datetime.strptime(entry['periodEnd'], PERIOD_END_AT_FORMAT).timestamp(),
+                    ))
+            except (KeyError, ValueError) as e:
+                msg = str(e)
+                log.error(
+                    'Error processing AdEx Tom pool incentives.',
+                    entry=entry,
+                    error=msg,
+                )
+                self.msg_aggregator.add_error(
+                    'Failed to deserialize the AdEx Tom pool incentives. '
+                    'Check logs for details. Ignoring it.',
+                )
+                return None
+
+        now = Timestamp(int(datetime.utcnow().timestamp()))
+        # Calculate Tom pool APR (APY)
```

Comment:
Is it APR, or APY? I think from our discussion it was APR. So keep one definition here only.